src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BackpackListener.java

```diff
@@ -103,6 +103,14 @@ public void onClick(InventoryClickEvent e) {
                             e.setCancelled(true);
                         }
                     }
+                } else if (e.getClick() == ClickType.SWAP_OFFHAND) { // Fixes #3265
+                    if (e.getClickedInventory().getType() != InventoryType.PLAYER) {
+                        ItemStack offHandItem = e.getWhoClicked().getInventory().getItemInOffHand();
+
+                        if (!isAllowed((SlimefunBackpack) backpack, offHandItem)) {
+                            e.setCancelled(true);
+                        }
+                    }
```

Comment:
Should this just always be cancelled? From what I can tell that isAllowed will be true still for the reported issue. Relevant code: https://github.com/Slimefun/Slimefun4/blob/eec5d2d66b2185b60c8eb34b293a9f8d78b6078a/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BackpackListener.java#L121-L127

```suggestion
                } else if (e.getClick() == ClickType.SWAP_OFFHAND && e.getClickedInventory().getType() != InventoryType.PLAYER) {
                    // Fixes #3265
                    ItemStack offHandItem = e.getWhoClicked().getInventory().getItemInOffHand();

                    if (!isAllowed((SlimefunBackpack) backpack, offHandItem)) {
                        e.setCancelled(true);
                    }
```

Reply:
Oh wait I see, the item will be the backpack which isn't allowed. Me dumb